doc/glossary.md

```diff
@@ -1,7 +1,56 @@
 # Definitions
 
-- 
+Basic Model Interface (BMI) - A library specification created by the Community Surface Dynamics Modeling System (CSDMS) to facilitate the conversion of a model or dataset into a reusable, plug-and-play component. [See BMI](https://bmi.readthedocs.io/en/latest/)
+
+Catchment - Generally, a physiographic unit where hydrologic processes take place. This class denotes a physiographic unit, which is defined by a hydrologically determined outlet to which all waters flow. While a catchment exists, it may or may not be clearly identified for repeated study. Specifically for this framework, a catchment represents an arbitrary spatial area and is an abstraction used to encapsulate a model. Every catchment has a catchment area realization. Every catchment of degree 2 will have a flowpath realization. [See HY\_Features](https://docs.opengeospatial.org/is/14-111r6/14-111r6.html#_catchment)
+
+Catchment Area Realization - A catchment implementation as a local water budget model which takes inputs from the atmosphere and non-surficial hydro geologic systems and contributes outputs to an outlet nexus.
+
+Catchment Aggregate - A set of non-overlapping dendritic and interior catchments arranged in an encompassing catchment. [See HY\_Features](https://docs.opengeospatial.org/is/14-111r6/14-111r6.html#_catchment_aggregate)
+
+Catchment Network - Edge-node topology pattern of a set of catchments connected by their hydro nexuses. [see HY\_Features](https://docs.opengeospatial.org/is/14-111r6/14-111r6.html#_catchment_network_topology)
+
+Cohesion - A measure of how well components of a software module fit together. Cohesion promotes readability, reliability, reusability, and robustness.
+
+Complex Realization - A single catchment representation of a network of higher detail catchement realizations and their relationships. This allows the modeled area to be represented at multiple levels of detail and supports dynamic high resolution nesting.
+
+Encapsulation - Bundling of hydrologic data and methods into a model for a given catchment realization.
+
+Formulation - The actualization of a catchment realization into a computable object with set parameters.
```

Comment:
This definitely needs expanded from they modeling perspective.  From the software perspective, we illustrate this as the interface.

Reply:
I am having a hard time with the formulation definition encompassing all of the latest discussions. @robertbartel do you mind taking a shot at this one?